test_api.py

```diff
@@ -1467,6 +1480,83 @@ def test_query_with_no_matching_insights_id(self):
+            assert host["id"] in host_id_list
+
+        self._post_paging_test(SEARCH_URL, query_doc, len(self.added_hosts))
+
+    def test_query_using_host_id_list_with_invalid_uuid(self):
+        host_list = self.added_hosts
+        host_id_list = [host.id for host in host_list]
+        host_id_list.append("notauuid")
+        query_doc = {"host_id_list": host_id_list}
+        self.post(SEARCH_URL, query_doc, 400)
+
+    def test_query_with_invalid_host_id_list(self):
+        invalid_query_docs = [None,
+                              {},
+                              {"host_id_list": []},
+                              {"host_id_list": {}}
+                              ]
+
+        for invalid_query_doc in invalid_query_docs:
+            with self.subTest(invalid_query_doc=invalid_query_doc):
+                self.post(SEARCH_URL,
+                          invalid_query_doc, 400)
+
+    def test_query_with_exclude_facts(self):
+        host_list = self.added_hosts
+        host_id_list = [host.id for host in host_list]
+        query_doc = {"host_id_list": host_id_list, "exclude_fields": ["facts"]}
+        response = self.post(SEARCH_URL, query_doc, 200)
+
+        for host in response["results"]:
+            assert host["id"] in host_id_list
+            assert "facts" not in host
+
+        self._post_paging_test(SEARCH_URL, query_doc, len(self.added_hosts))
+
+    def test_query_with_exclude_system_profile(self):
+        host_list = self.added_hosts
+        host_id_list = [host.id for host in host_list]
+        query_doc = {"host_id_list": host_id_list, "exclude_fields": ["system_profile_facts"]}
+        response = self.post(SEARCH_URL, query_doc, 200)
+
+        for host in response["results"]:
+            assert host["id"] in host_id_list
+            assert "facts" in host
+            assert "system_profile_facts" not in host
+
+        self._post_paging_test(SEARCH_URL, query_doc, len(self.added_hosts))
+
+    def test_query_with_exclude_facts_and_system_profile(self):
+        host_list = self.added_hosts
+        host_id_list = [host.id for host in host_list]
+        query_doc = {"host_id_list": host_id_list, "exclude_fields": ["facts", "system_profile_facts"]}
+        response = self.post(SEARCH_URL, query_doc, 200)
+
+        for host in response["results"]:
+            assert host["id"] in host_id_list
+            assert "facts" not in host
+            assert "system_profile_facts" not in host
+
+        self._post_paging_test(SEARCH_URL, query_doc, len(self.added_hosts))
+
+    def test_query_with_invalid_exclude(self):
+        host_list = self.added_hosts
+        host_id_list = [host.id for host in host_list]
+        query_doc = {"host_id_list": host_id_list, "exclude_fields": ["invalid"]}
+        response = self.post(SEARCH_URL, query_doc, 400)
+        assert "Invalid exclude_fields. Valid values are: ['facts', 'system_profile_facts']" in response["detail"]["exclude_fields"][0]
```

Comment:
This line is too long.
```suggestion
        self.assertIn("Invalid exclude_fields. Valid values are: ['facts', 'system_profile_facts']", response["detail"]["exclude_fields"][0])
```